### PR TITLE
Add support for 'any resource' codegen

### PR DESCRIPTION
**Pull request by AaronFriel**
Part of pulumi/pulumi#6346.

Depends on commit in https://github.com/pulumi/pulumi/pull/15793.

### CODE REVIEW

File: pkg/codegen/java/names/known.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var PulumiInternal = PulumiCore.Dot("internal")
 
 var PulumiAnnotations = PulumiCore.Dot("annotations")
 
-var ResourceType = PulumiAnnotations.Dot("ResourceType")
+var ResourceTypeAnnotation = PulumiAnnotations.Dot("ResourceType")
```
Comment (t0yv0) [inline pkg/codegen/java/names/known.go:23]: Very nice, less confusing.

**Review by t0yv0 — APPROVED** [review]
This looks good, fairly surgical. For good form it would be nice to see a quick test and have the go.mod upgrade separate but I don't know how much friction there is in this repo. Thank you!

**Comment by AaronFriel** [comment]
@t0yv0 @justinvp getting tests to pass required more changes than I expected, as it looks like SDK codegen has changed:

https://github.com/pulumi/pulumi-java/pull/1336/files#diff-5faecbaf7ae0e55d4c63d83d70d5144ba4fd1dcf5eadfd47260a3f5de57fab16

But tests are passing. I leave this up to you to review & decide if this is mergeable.

**Review by t0yv0 — APPROVED** [review]
LGTM but I don't do much in terms of maintenance on this repo these days.. Looks like a surgical change plus lots of go.mod update-related Chun in testdata.

**Pull request merged**